Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -1039,6 +1028,48 @@ int MXC_SPI_SlaveTransactionAsync(mxc_spi_req_t *req);
  */
 int MXC_SPI_SlaveTransactionDMA(mxc_spi_req_t *req);
 
+/**
+ * @brief   Setup an interrupt-driven, non-blocking SPI Target transaction.
+ * 
+ * @param   spi         Pointer to SPI instance's registers.
+ * @param   tx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   tx_fr_len   Number of frames to transmit from transmit buffer.
+ * @param   rx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   rx_fr_len   Number of frames to store in recieve buffer.
+ *
+ * @return  See \ref MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_SPI_TargetTransaction(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
```

Comment:
I don't think it makes sense to make this non-blocking.  `SPI_TargetTransaction` should be a replacement for `SPI_MasterTransaction`, which is blocking.

Reply:
Are you referring to SPI_SlaveTransaction instead? (Master => Controller :: Slave => Target).

Reply:
Yes, sorry.  Same point though.  `SPI_SlaveTransaction` was blocking previously